patch-db/src/locker/mod.rs

```diff
@@ -117,10 +153,67 @@ impl Drop for CancelGuard {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+enum LockInfos {
+    LockInfo(LockInfo),
+    LockInfos(Vec<LockInfo>),
+}
+impl LockInfos {
+    fn conflicts_with(&self, other: &LockInfos) -> bool {
+        match (self, other) {
+            (LockInfos::LockInfo(lock_info), LockInfos::LockInfo(other_lock_info)) => {
+                lock_info.conflicts_with(other_lock_info)
+            }
+            (LockInfos::LockInfo(lock_info), LockInfos::LockInfos(other_lock_infos)) => {
+                other_lock_infos
+                    .iter()
+                    .any(|other_lock_info| lock_info.conflicts_with(other_lock_info))
+            }
+            (LockInfos::LockInfos(lock_infos), LockInfos::LockInfo(other_lock_info)) => lock_infos
+                .iter()
+                .any(|lock_info| lock_info.conflicts_with(other_lock_info)),
+            (LockInfos::LockInfos(lock_infos), LockInfos::LockInfos(other_lock_infos)) => {
+                lock_infos.iter().any(|lock_info| {
+                    other_lock_infos
+                        .iter()
+                        .any(|other_lock_info| lock_info.conflicts_with(other_lock_info))
+                })
+            }
+        }
+    }
+
+    fn as_vec(&self) -> Vec<&LockInfo> {
```

Comment:
Not sure how I feel about this, but this is the solution that I am using. Was originally going to return an iterator, but having the matching made it so each would be a different branch therefore became a problem with the same iterator.